### PR TITLE
Readd file associations

### LINKED ISSUE #5520: Add file associations when installing
Latest development version does not associate bib files with JabRef (using Windows installer, but probably applies to Linux and Mac as well).

jpackage has the following option:
> --file-associations <file path> — Path to a Properties file that contains list of key, value pairs (absolute path or relative to the current directory). The keys "extension", "mime-type", "icon", and "description" can be used to describe the association. This option can be used multiple times.

The documentation is a bit sparse at this point. Probably needs a bit of experimentation using different formats. Relevant part of the build script: https://github.com/JabRef/jabref/blob/9fd49bcb2560a46c7f319dd59fdd9a58845415ce/build.gradle#L565-L607

### PR DESCRIPTION
**Pull request by tobiasdiez**
Fixes #5520. At least works on Windows. @LyzardKing could you please test this for linux as well? Thanks!

----

- [ ] Change in CHANGELOG.md described
- [ ] Tests created for changes
- [ ] Manually tested changed features in running JabRef
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

### CODE REVIEW

**Comment by LyzardKing** [comment]
I'll test it and see if the mimetype line can be removed from the .desktop template

**Comment by LyzardKing** [comment]
@tobiasdiez It works, in adding the association.
The problem is that jpackage isn't using the altered .desktop template in the buildres folder, 
and that (I think) is also the reason for the issue with the browser addon opening new instances...
We'ew not adding the following:
```
GenericName=BibTeX Editor
Keywords=bibtex;biblatex;latex;bibliography
StartupWMClass=org-jabref-JabRefMain
```

**Comment by LyzardKing** [comment]
I think I found out how to fix the issue.
If you want to add the change to this PR the diff is minimal:
- rename `buildres/linux/jabref-JabRef.desktop` to `buildres/linux/JabRef.desktop`
- in that file replace the MimeType line with  `DESKTOP_MIMES`

**Comment by koppor** [comment]
@LyzardKing I hope, I understood your comment correctly: https://github.com/JabRef/jabref/pull/5625/commits/1f607e5069b40b3b6add89e3fd9dc5ca270194bc

**Comment by tobiasdiez** [comment]
Maybe the icon name in the association file is wrong. Do you have a png file next to the executable in the installed directory?

**Comment by LyzardKing** [comment]
@koppor Yes, that's the same thing I tested..
I'll have a look at the deb on master.

File: buildres/linux/JabRef.desktop [hunk 1]
```diff
@@ -6,7 +6,7 @@ Exec=APPLICATION_LAUNCHER
 Icon=APPLICATION_ICON
 Terminal=false
 Type=Application
-MimeType=text/x-bibtex;
+MimeType=DESKTOP_MIMES
```
Comment (LyzardKing) [inline buildres/linux/JabRef.desktop]:
```suggestion
DESKTOP_MIMES
```

Comment (LyzardKing) [reply buildres/linux/JabRef.desktop]: This is the only other change

**Comment by LyzardKing** [comment]
@tobiasdiez in jabref/lib there's the icon, and it's `JabRef.png`
But it doesn't seem to work with that either..

**Pull request merged**